sql/core/src/test/scala/org/apache/spark/sql/SQLInsertTestSuite.scala

```diff
@@ -160,6 +160,42 @@ trait SQLInsertTestSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("insert overwrite with column list - by name") {
```

Comment:
since we are doing followup, can we add more negative end-to-end tests? like when the name does not match.